packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_CoarseOperator_def.hpp

```diff
@@ -664,6 +734,158 @@ namespace FROSch {
+                MLnumMyRows = 0;
+                numProcsGatheringStep = LO(numProcsGatheringStep/MLgatheringFactor);
+                if (this->MpiComm_->getRank()%(this->MpiComm_->getSize()/numProcsGatheringStep) == 0 && this->MpiComm_->getRank()/(this->MpiComm_->getSize()/numProcsGatheringStep) < numProcsGatheringStep) {
+                    if (this->MpiComm_->getRank()==0) {
+                        MLnumMyRows = MLnumGlobalIndices - (MLnumGlobalIndices/numProcsGatheringStep)*(numProcsGatheringStep-1);
+                    } else {
+                        MLnumMyRows = MLnumGlobalIndices/numProcsGatheringStep;
+                    }
+                }
+                MLGatheringMaps_[i] = Xpetra::MapFactory<LO,GO,NO>::Build(CoarseMap_->lib(),-1,MLnumMyRows,0,this->MpiComm_);
+            }
+
+            MLnumMyRows = 0;
+            if (this->MpiComm_->getRank()%(this->MpiComm_->getSize()/NumProcsCoarseSolve_) == 0 && this->MpiComm_->getRank()/(this->MpiComm_->getSize()/NumProcsCoarseSolve_) < NumProcsCoarseSolve_) {
+              if (this->MpiComm_->getRank()==0) {
+                  MLnumMyRows = MLnumGlobalIndices - (MLnumGlobalIndices/NumProcsCoarseSolve_)*(NumProcsCoarseSolve_-1);
+              } else {
+                  MLnumMyRows = MLnumGlobalIndices/NumProcsCoarseSolve_;
+              }
+            }
+            MLGatheringMaps_[MLgatheringSteps-1] = Xpetra::MapFactory<LO,GO,NO>::Build(CoarseMap_->lib(),-1,MLnumMyRows,0,this->MpiComm_);
+            for (UN j=1; j<MLGatheringMaps_.size(); j++) {
+              MLCoarseSolveExporters_[j] = Xpetra::ExportFactory<LO,GO,NO>::Build(MLGatheringMaps_[j-1],MLGatheringMaps_[j]);
+            }
+            int nSubs = this->MpiComm_->getSize();
+            GOVec RowsCoarseSolve;
+						if (OnCoarseSolveComm_) {
+									 int start = (nSubs*(CoarseSolveComm_->getRank()))/NumProcsCoarseSolve_;
+									 int end = (nSubs*(CoarseSolveComm_->getRank()+1))/NumProcsCoarseSolve_;
+									 RowsCoarseSolve.resize(end-start);
+									 for (int i = 0; i<end-start; i++) {
+											 RowsCoarseSolve[i] = start+i;
+									 }
+					   }
+
+						 MLCoarseMap_ = Xpetra::MapFactory<LO,GO,NO>::Build(CoarseMap_->lib(),-1,RowsCoarseSolve,0,CoarseSolveComm_);
+
+             //#####################################################################
+             // Build Repeated Map Zoltan2
+             // build ElementNodeList_ to have adjacent entities to one subdomain
+             this->buildElementNodeList();
+             // Connectivity Graph on the CoarseSolveComm_
+             this->buildCoarseGraph();
+             //Build Repeatd Map on CoarseComm------------
+             //Initialize Maps...
+             ConstXMapPtr UniqueMap;
+             XMapPtr UniqueMapAll;
+             XMapPtr tmpRepMap;
+             ConstXMapPtr ConstRepMap;
+             GOVec uniEle;
+
+             if(OnCoarseSolveComm_){
+               //Coarse DofsMaps so far only one Block will work
```

Comment:
Is this checked somewhere beforehand, so a block problem can't reach this point?

Reply:
I haven checked this yet.

Reply:
I added an ASSERT in `IPOUHarmonicCoarseOperator`